### ISSUE TITLE
Could FCOS overfit one single train image

### ISSUE THREAD

**Issue by LouieYang** [tianzhi0549/FCOS#20]
Hi, I am trying to overfit one single image using FCOS (train/test with single image without any other transforms like horizontal flip) to test if I correctly use your codes. I have a very strange result and it seems like it can not fully overfit one single training images. 

Inference: 
![prediction|center|200x0](https://user-images.githubusercontent.com/10849611/56858101-8bf42900-69a7-11e9-8968-7a71bc7be28e.png)

Ground-truth:
![bounding_box](https://user-images.githubusercontent.com/10849611/56858120-c067e500-69a7-11e9-8d62-775e2070d9bf.png)

I have borrowed codes from mask rcnn repo and use the fcos code from rpn/fcos but didn't check other differences between these two repos. **Am I missing something that may cause this problem?**

**or**

**FCOS just can't fully overfit one single image like Mask RCNN** because it uses multiple binary classifier (Sigmoid Focal Loss) instead of SoftmaxFocalLoss so that other classes (except for instances that appear on training image) classifier won't train for single image dataset. 

Could you give me some hints on debugging this problem?

Many thanks.

**Comment by tianzhi0549** [comment]
@LouieYang It might be because of GNs in tower layers. Please remove all GN layers there and try again.

**Comment by LouieYang** [comment]
Thank you @tianzhi0549, the inference result doesn't change after I remove the GN layers. 

I finally fix it by increasing INFERENCE_TH to 0.2 and it can overfit one image now. Still wondering why those boxes appear with 0.05 < scores < 0.2, any idea of it ?

**Comment by tianzhi0549** [comment]
@LouieYang I am not sure. It might be because we directly output the top 100 predictions after the post-processing, regardless of their scores.

**Comment by LouieYang** [comment]
@tianzhi0549 Seems like in [inference](https://github.com/tianzhi0549/FCOS/blob/de2d65a5e1ad01f0db2e16fdd4be682fd18a553b/maskrcnn_benchmark/modeling/rpn/fcos/inference.py#L175), fcos filtered top 100 predictions according to their scores. 

Thanks again for the nice code btw. 
Closed (FCOS could correctly overfit one training sample)

**Comment by LMerCy** [comment]
> Thank you @tianzhi0549, the inference result doesn't change after I remove the GN layers.
> 
> I finally fix it by increasing INFERENCE_TH to 0.2 and it can overfit one image now. Still wondering why those boxes appear with 0.05 < scores < 0.2, any idea of it ?

Hello, i also meet the same problem in mmdet version of fcos, i train the model with pytorch QAT.
